compiler/ilgen/OMRIlBuilder.hpp

```diff
@@ -108,9 +120,13 @@ class IlBuilder : public TR::IlInjector
          JBCase(int32_t v, TR::IlBuilder *b, int32_t f)
              : _value(v), _builder(b), _fallsThrough(f) {}
 
+      private:
          int32_t _value;          // value matched by the case
          TR::IlBuilder *_builder; // builder for the case body
          int32_t _fallsThrough;   // whether the case falls-through
+         void * _client;
+         static ClientAllocator _clientAllocator;
+         static ImplGetter _getImpl;
```

Comment:
I haven't run into any technical challenges that would make using an ImplGetter a requirement..

While I agree that using function pointers is anything but elegant, I think that mandating that `_impl` be the first field is more confusing. Having to access the field while avoiding strict-aliasing problems and undefined behaviour makes it more difficult to reason about correctness. Using an ImplGetter callback, the only real complexity is in how the field gets assigned. Beyond that, it can be treated like a regular getter function (perhaps requiring an extra null check).

Also, I don't know yet how enforcing that `_impl` be the first field might affect binding to languages other than C and C++.

Reply:
All right, I won't fight it since it works and provides a degree of flexibility.

Having fields adhere to a defined layout may seem "confusing" from a cleanliness point of view (though I don't really want to concede that point), but it can sure come in handy when you're debugging something without symbols.